code/modules/mod/modules/modules_general.dm

```diff
@@ -571,10 +571,13 @@
 	. = ..()
 	on_emag(src, user, emag_card)
 
-/obj/item/mod/module/dna_lock/proc/dna_check()
-	if(!dna || (mod.wearer.has_dna() && mod.wearer.dna.unique_enzymes == dna))
+/obj/item/mod/module/dna_lock/proc/dna_check(mob/user)
+	if(!user && !iscarbon(user))
+		return FALSE
+	var/mob/living/carbon/C = user
+	if(!dna  || (C.has_dna() && C.dna.unique_enzymes == dna))
 		return TRUE
-	balloon_alert(mod.wearer, "dna locked!")
+	balloon_alert(user, "dna locked!")
```

Comment:
Better var name here. something like `carbon_user` instead of `C`

Reply:
Names are hard, suggestions put in